waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/DatabaseWhitelistAccessControlHandler.java

```diff
@@ -41,15 +45,30 @@ public DatabaseWhitelistAccessControlHandler(PrimaryMetaStore primaryMetaStore,
   }
 
   private void add(String databaseName) {
-    databaseWhiteList.add(databaseName.trim().toLowerCase());
+    databaseWhiteList.add(Pattern.compile(trimToLowerCase(databaseName)));
+  }
+
+  private String trimToLowerCase(String string) {
+    return string.trim().toLowerCase();
   }
 
   @Override
   public boolean hasWritePermission(String databaseName) {
     if (databaseName == null) {
       return true;
     }
-    return databaseWhiteList.contains(databaseName.trim().toLowerCase());
+    databaseName = trimToLowerCase(databaseName);
+    if (databaseWhiteList.contains(databaseName)) {
```

Comment:
Check whether this if statement is ever returning true for tests, I suspect not and if so then it can be removed